src/extensions/install-framework.js

```diff
@@ -20,43 +20,47 @@ module.exports = (toolbox) => {
 
     muted(`Installing ${framework}...`);
 
-    pkgJson.dependencies[framework] = frameworkVersion[framework];
+    Object.assign(pkgJson.dependencies, {
+      [framework]: frameworkVersion[framework],
+    });
 
+    // dotenv
+    Object.assign(pkgJson.devDependencies, {
+      dotenv: '^16.0.3',
+    });
+
+    copyAsync(`${assetsPath}/dotenv/.env.example`, `${appDir}/.env.example`);
+
+    // TypeScript
+    if (projectLanguage === 'TS') {
+      pkgJson.scripts['start'] = 'node -r dotenv/config dist/index';
+
+      await Promise.all([
+        copyAsync(`${assetsPath}/src/`, `${appDir}/src/`),
+        copyAsync(`${assetsPath}/test/`, `${appDir}/test/`),
+      ]);
+    }
+
+    // Express
     if (framework === 'express') {
       Object.assign(pkgJson.dependencies, {
         helmet: '^6.0.1',
         cors: '^2.8.5',
         compression: '^1.7.4',
       });
-    }
 
-    try {
+      // with TypeScript
       if (projectLanguage === 'TS') {
-        await Promise.all([
-          copyAsync(`${assetsPath}/src/`, `${appDir}/src/`),
-          copyAsync(`${assetsPath}/test/`, `${appDir}/test/`),
-        ]);
-
-        pkgJson.scripts['start'] = 'node dist/index';
-
-        if (framework === 'express') {
-          copyAsync(`${assetsPath}/${framework}/src/`, `${appDir}/src/`, {
-            overwrite: true,
-          });
-        }
-      }
+        copyAsync(`${assetsPath}/${framework}/src/`, `${appDir}/src/`, {
```

Comment:
IMO we can structure this a bit better. If I follow correctly - we copy a standard `src/` folder and then if the case is Express with TypeScript we copy the "special" `framework/src/` folder. If that's the case we can:
```
// ...setup for all cases


if (framework === 'express') {
  // ...setup for express case
  if (projectLanguage === 'TS') {
    // ...copy and setup for express + TS case
    success(...)
    return
  }
}
// copy for other common cases
success(...)

```

Reply:
Since it's only one file it doesn't really matter, so I'm approving :)